src/Stratis.Bitcoin.Features.Wallet/WalletTransactionHandler.cs

```diff
@@ -98,9 +107,17 @@ public void FundTransaction(TransactionBuildContext context, Transaction transac
 
             foreach (TxIn transactionInput in transaction.Inputs)
                 context.SelectedInputs.Add(transactionInput.PrevOut);
+        }
 
-            Transaction newTransaction = this.BuildTransaction(context);
-
+        /// <summary>
+        /// In the case of funding a <see cref="Transaction"/>,  ensures all the features of the transaction are transferred over from 
+        /// a transaction that was built via <see cref="BuildTransactionInternal(TransactionBuildContext)"/>.
+        /// </summary>
+        /// <param name="transaction"></param>
+        /// <param name="newTransaction"></param>
+        /// <param name="context"></param>
+        private void UpdateTransactionFromBuiltTransaction(Transaction transaction, Transaction newTransaction, TransactionBuildContext context)
+        {
             if (context.ChangeAddress != null)
             {
                 // find the position of the change and move it over.
```

Comment:
If I remember correctly, this movement of the change address is only done so that the masternode registration protocol knows which address is the change. It essentially defeats the point of having shuffled outputs in the first place. The change address position could be an enum instead (First, Last, Random)? Or just an int for position, but I'm not sure if that's worth the possibility of error in the caller.